scripts/local/utils/bulk_ops_common.py

```diff
@@ -119,3 +121,38 @@ def run_argparser(argparser):
         raise Exception(f"\nError: Unknown arguments: {unknown}")
 
     return args
+
+
+def resize_image(
+    input_path, output_path, max_width=None, max_height=None, trigger_size=0
+):
+    """
+    Resize an image based on given width, height, and trigger size.
+
+    :param input_path: Path to the input image.
+    :param output_path: Path to save the resized image.
+    :param max_width: Maximum width for resizing.
+    :param max_height: Maximum height for resizing.
+    :param trigger_size: Minimum file size in bytes to trigger resizing.
+    """
+    try:
+        if os.path.getsize(input_path) > trigger_size:
+            with Image.open(input_path) as img:
+                width, height = img.size
+
+                # Resize based on max width or height while keeping the aspect ratio
+                if max_width and width > max_width:
+                    new_height = int((max_width / width) * height)
+                    img = img.resize((max_width, new_height), Image.ANTIALIAS)
+
+                if max_height and height > max_height:
+                    new_width = int((max_height / height) * width)
+                    img = img.resize((new_width, max_height), Image.ANTIALIAS)
```

Comment:
I already have a util created for this in image_utils.py, can you check if we can use that?